src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -615,11 +679,25 @@ public PublisherClientImpl(Publisher.IPublisherClient grpcClient, PublisherSetti
             };
             return s_listTopicsPageStreamer.FetchAsync(
                 request,
-                (pageStreamRequest, cancellationToken) => GrpcClient.ListTopicsAsync(
-                    pageStreamRequest,
-                    _clientHelper.BuildCallOptions(cancellationToken, callSettings)
-                ).ResponseAsync
-            );
+                (pageStreamRequest, cancellationToken) =>
+                {
+                    // if IAsyncEnumerator.MoveNext(CancellationToken) is passed a default cancellation-token
+                    // (also CancellationToken.None), then use the CancellationToken provided in callSettings.
+                    // Otherwise use the MoveNext() cancellation-token.
+                    // It's unfortunate (for us!) that MoveNext() takes a CancellationToken, rather than a
+                    // nullable CancellationToken?
+                    CallSettings useCallSettings;
```

Comment:
We could potentially pass in something which will be cancelled if *either* of the cancellation tokens involved is cancelled.